### PR TITLE
[BE] Added strict type checking. Minor refactor to types and code

### PR DESCRIPTION
**Pull request by stonefruit**
Add `"strict": true` to tsconfig for improved type safety for backend. Resolved associated type issues.

### CODE REVIEW

File: backend/src/app.ts [hunk 1]
```diff
@@ -15,10 +15,13 @@ import { ApiErrorMiddleware } from './utils/error';
 import proxy from "express-http-proxy";
 
 // Handle Express req user
-declare module 'express' {
+declare module 'express-serve-static-core' {
```
Comment (stonefruit) [inline backend/src/app.ts:18]: Express object now comes from here.

https://stackoverflow.com/a/55718334/2085381

File: backend/src/config/passport.ts [hunk 2]
```diff
@@ -7,13 +7,17 @@ import { User, UserCreationAttributes } from "../models/user";
 const authStrategy = (passport: passport.PassportStatic): void => {
   const LocalStrategy = passportLocal.Strategy;
   // serialize
-  passport.serializeUser((serialUser: User, done) => {
-      done(null, serialUser.id);
+  passport.serializeUser((_user, done) => {
+      const user = _user as unknown as User
```
Comment (stonefruit) [inline backend/src/config/passport.ts:11]: passport uses Express.User, which is incompatible with our sequelize User, hence retyped to allow strict type checking to pass

File: backend/src/controllers/upload.ts [hunk 3]
```diff
@@ -67,19 +67,18 @@ class UploadController {
...
         }
-      }).then((foundUpload: Upload) => {
-        if (foundUpload) {
-          filename = foundUpload.filename;
-        } else {
-          const result = {"status": "failure", "message": "File ID does not exist"};
-          return res.end(JSON.stringify(result));
-        }
-      });
+    })
```
Comment (stonefruit) [inline backend/src/controllers/upload.ts:75]: minor refactor to avoid mixed async/await and promises. also helps to resolve typescript errors.

**Review by telboon — APPROVED** [review]
Thanks for PR, and thanks for the comments to go along with the changes, making the changes clear. Cheers!

**Pull request merged**